docs/tutorials/tabular-fasttext.md

```diff
@@ -0,0 +1,219 @@
+    4             0              40     El-Salvador  <=50K  
+
+
+Note that we loaded data from a CSV file stored in the cloud (AWS s3 bucket), but you can you specify a local file-path instead if you have already downloaded the CSV file to your own machine (e.g., using `wget`).
+Each row in the table `train_data` corresponds to a single training example. In this particular dataset, each row corresponds to an individual person, and the columns contain various characteristics reported during a census.
+
+Let's first use these features to predict whether the person's income exceeds $50,000 or not, which is recorded in the `class` column of this table.
+
+
+```python
+label_column = 'class'
+print("Summary of class variable: \n", train_data[label_column].describe())
+```
+
+    Summary of class variable: 
+     count     39073
+    unique        2
+    top       <=50K
+    freq      29704
+    Name: class, dtype: object
+
+
+# Train FastText model alone
+
+
+```python
+X_train = train_data.drop(columns=[label_column])
+y_train = train_data[label_column]
+
+problem_type = infer_problem_type(y=y_train)  # Infer problem type (or else specify directly)
+
+fasttext_model = FastTextModel(path='fasttext-model/', name='FastTextModel', problem_type=problem_type,
+                              hyperparameters={'epoch': 50})
+fasttext_model.fit(X_train=X_train, y_train=y_train)
+```
+
+    Here are the 2 unique label values in your data:  ['<=50K', '>50K']
+    AutoGluon infers your prediction problem is: binary  (because only two unique label-values observed).
+    If this is wrong, please specify `problem_type` argument in fit() instead (You may specify problem_type as one of: ['binary', 'multiclass', 'regression'])
+    
+    Model FastTextModel's eval_metric inferred to be 'accuracy' because problem_type='binary' and eval_metric was not specified during init.
+    NLP features ['workclass', 'education', 'marital-status', 'occupation', 'relationship', 'race', 'sex', 'native-country']
+
+
+
+```python
+y_pred = fasttext_model.predict(test_data)
+
+df_res = pd.DataFrame({
+    'pred': y_pred,
+    'label': test_data[label_column]
+})
+print((df_res.pred.str.strip() == df_res.label.str.strip()).value_counts())
+print(df_res.sample(5))
+```
+
+    True     8218
+    False    1551
+    dtype: int64
+           pred   label
+    2028  <=50K    >50K
+    7243  <=50K   <=50K
+    7704  <=50K   <=50K
+    208   <=50K   <=50K
+    6429  <=50K   <=50K
+
+
+# Use fastText model in TabularPrediction task 
+
+
+```python
+custom_hyperparameters = {'RF': {},
+                         FastTextModel:  {'epoch': 50}
+                         }
+# mock some NLP fields
+train_data['text'] = (
+    test_data[['education', 'marital-status', 'occupation', 'relationship']]
+    .apply(lambda r: ' '.join(r.values), axis=1)
+)
+
+test_data['text'] = (
+    test_data[['education', 'marital-status', 'occupation', 'relationship']]
```

Comment:
Consider trying more predictive feature-set, like say:
['education', 'occupation', 'workclass', 'native-country']

Reply:
I'd also add some dummy punctation at the end or when concatenating two of the columns, just to show the fastText preprocessing can properly handle this.

Also add:
`print(train_data['text'].head())`
to show user what this `text` field looks like.